### DISCUSSION TITLE
cURL timeout looking for Firefly updates

### DISCUSSION THREAD

**Discussion by Kaemmelot** [firefly-iii/firefly-iii#5713] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been discussed before.

### Description

As #4004 mentioned earlier, I ran into the same issue:
> Guzzle: cURL error 28: Resolving timed out after 3141 milliseconds (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://version.firefly-iii.org/index.json

I tested this in the console and logged the time it takes to establish the connection:
> curl -w "\n\n%{time_namelookup}\n%{time_total}\n" https://version.firefly-iii.org/index.json
> [...]
> 5.036663
> 5.235301

So at the moment the DNS lookup for this domain seems to take 5 seconds. I couldn't find any other domain that takes this long, but I think it might be because of a cloudflare protection (which I found in the HTTP response headers).

### Solution

Could you either increase the timeout or make it configurable for users facing this problem?

**Comment by JC5** [comment]
Mmmm, that's weird.

I can't imagine Cloudflare of all places to take long with DNS lookups. I'm getting results in the ~0.02 range myself, several magnitudes faster, from different places around the world. 

If it's not working in 3 seconds something else is happening outside of my control.
